src/dbos-executor.ts

```diff
@@ -92,10 +100,12 @@ export const OperationType = {
   WORKFLOW: "workflow",
   TRANSACTION: "transaction",
   COMMUNICATOR: "communicator",
+  PROCEDURE: "procedure",
 } as const;
 
 const TempWorkflowType = {
   transaction: "transaction",
+  procedure: "procedure",
```

Comment:
There are several places where we check `TempWorkflowType.transaction`. For example:
```
if (wCtxt.tempWfOperationType !== TempWorkflowType.transaction) {
      args = await this.systemDatabase.initWorkflowStatus(internalStatus, args);
    }
```
And
```
if (wCtxt.tempWfOperationType === TempWorkflowType.transaction) {
          // For single-transaction workflows, asynchronously record inputs.
          // We must buffer inputs after workflow status is buffered/flushed because workflow_inputs table has a foreign key reference to the workflow_status table.
          this.systemDatabase.bufferWorkflowInputs(workflowUUID, args);
        }
```
These checks eliminate synchronous writes to the system database when we execute a single-transaction workflow. I think we also want to do that for stored procs.